app/models/user.rb

```diff
@@ -424,6 +424,14 @@ def guard_unconfirmed_email
     end
   end
 
+  def reorder_aspects(aspect_order)
+    i = 0
+    aspect_order.each do |id|
+      self.aspects.find(id).update_attributes({ :order_id => i })
+      i += 1
+    end
+  end
```

Comment:
I'm not sure about the JS part, somebody else has to check that, but I think this should get its own route on `AspectsController` and this method can be nicer written by using `each_with_index`. There's also no need for the `self.` and I'd name the parameter `ordered_aspect_ids`.